arangod/Replication2/StateMachines/Document/DocumentStateShardHandler.cpp

```diff
@@ -145,19 +145,31 @@ auto DocumentStateShardHandler::getShardMap() -> ShardMap {
 auto DocumentStateShardHandler::ensureIndex(
     ShardID shard, std::shared_ptr<VPackBuilder> const& properties,
     std::shared_ptr<methods::Indexes::ProgressTracker> progress) -> Result {
-  auto res = basics::catchToResult([&] {
-    return _maintenance->executeCreateIndex(shard, properties,
-                                            std::move(progress));
+  auto res = basics::catchToResult(
+      [this, &properties, shard = std::move(shard),
+       progress = std::move(progress)]() mutable -> Result {
+        return _maintenance->executeCreateIndex(shard, properties,
+                                                std::move(progress));
+      });
+
+  if (res.ok()) {
+    _maintenance->addDirty();
+  }
+  return res;
+}
+
+auto DocumentStateShardHandler::dropIndex(ShardID shard,
+                                          velocypack::SharedSlice index)
+    -> Result {
+  auto res = basics::catchToResult([this, shard = std::move(shard),
+                                    index =
+                                        std::move(index)]() mutable -> Result {
+    return _maintenance->executeDropIndex(std::move(shard), std::move(index));
   });
 
-  if (res.fail()) {
-    res = Result{res.errorNumber(),
-                 fmt::format("Error: {} Failed to ensure index on shard {}: {}",
```

Comment:
Don't we loose this additional log information now?

I think this is important as the general Index error message is generic, and this adds the context on the shard.

Reply:
Addressed in https://github.com/arangodb/arangodb/pull/19760/commits/2493f257db51548e5dc338b416c957773ca59478

Reply:
:+1: